lke-configurations/alerts.cypher

```diff
@@ -1,56 +1,137 @@
-// /*
-//  * Name: UNNAMED #1
-//  * Desc: Find Mortgage Loans with a Realtor or a broker in a different state
-//  *
-//  * Target: m
-//  *
-//  * Custom columns:
-//  *   name   | Name    | text
-//  *   b.name | Broker  | text
-//  *   r.name | Realtor | text
-//  */
-
-MATCH (b:Company)<-[hb:HAS_BROKER]-(m:MortgageLoan)<-[hl:HAS_LOAN]-(p)
-MATCH (m)-[hr:HAS_REALTOR]->(r:Company)
-WITH m, p, b, r, hl, hb, hr, split(m.address, ",") as ma, split(b.address, ",") as ba, split(r.address, ",") as ra
-WHERE ma[size(ma)-1] <> ba[size(ba)-1] OR ma[size(ma)-1] <> ba[size(ra)-1]
-RETURN m, p, b, r, hl, hb, hr,  coalesce(p.full_name, p.name) as name, b.name, r.name
-
-
-
 // /*
 //  * Name: Common Clients
 //  * Desc: Find group of bank clients (more than 3) involved in Mortgage Loans that share the same Loaners and Brokers
 //  *
 //  * Target: r
 //  *
 //  * Custom columns:
-//  *   b.name | Broker        | text
-//  *   r.name | Realtor       | text
-//  *   len    | N. of clients | number
+//  *   Broker        | b.name | text
+//  *   Realtor       | r.name | text
+//  *   N. of clients | len    | number
 //  */
 
+
+// Model 1
 MATCH (p{is_client:true})-[hl:HAS_LOAN]->(m:MortgageLoan)
 MATCH (r:Company)<-[hr:HAS_REALTOR]-(m)-[hb:HAS_BROKER]->(b:Company)
-WITH r, b, collect(p) as col, collect(hl) + collect(hr) + collect(hb) as rel
+WITH r, b, collect(p) + collect(m) as col, collect(hl) + collect(hr) + collect(hb) as rel
 WHERE size(col) > 3
-RETURN r, b, size(col) as len, r.name, b.name,  col, rel
-
+RETURN r, b, col, rel
 
+// Query calculating case attributes
+MATCH (r:Company)<-[hr:HAS_REALTOR]-(m)-[hb:HAS_BROKER]->(b:Company)
```

Comment:
In principle the query is not well designed. The target of the alert is `r` (the Company) so you should ensure your case attribute is returning a single result for your company. In your case you are grouping by `r` and `b` causing (possibly) multiple results.

Also you are taking all the possible nodes between Companies (including things that may not be `MorgageLoan` as defined in the model and nodes that maybe are not part of the case for whatever reason). It would be better to put a filter `where m:MortgageLoan and id(m) in {{"context":nodeset}}`.

Also you are not really using the company `b` (except for the faulty count). Consider to avoid matching it if not relevant.

if you would like to get the list of broker names, consider concatenating them.
Note as internal mechanism of LKE, if you get as results
|Realtor|Broker|len|
|---|---|---|
|r1|b1|1|
|r1|b2|2|

In the columns you will get `r1` for Realtor, `b1, b2` for Brokers (LKE is automatically concatenating the strings), `2` for Len (LKE is taking the last number).

If you are focus on both, consider declaring `[r,b] as target`

Reply:
I changed the target to be r and b